src/components/FormInput/FormInput.test.tsx

```diff
@@ -0,0 +1,88 @@
+import userEvent from '@testing-library/user-event';
+import { useState } from 'react';
+import { describe, it, expect, vi } from 'vitest';
+
+import { render, screen } from '../../test/test-utils';
+
+import { FormInput } from './FormInput';
+
+describe('FormInput', () => {
+  const defaultProps = {
+    label: 'Mail Id',
+    value: '',
+    onChange: vi.fn(),
+    'data-testid': 'form-input',
+  };
+
+  const FormInputWrapper = ({ value: initialValue = '', onChange, ...props }) => {
+    const [value, setValue] = useState(initialValue);
+    const handleChange = (e) => {
+      setValue(e.target.value);
+      if (onChange) onChange(e);
+    };
+    return <FormInput {...props} value={value} onChange={handleChange} />;
+  };
+
+  it('renderiza con props por defecto', () => {
+    render(<FormInputWrapper {...defaultProps} />);
+    const inputContainer = screen.getByTestId('form-input');
+    expect(inputContainer).toBeInTheDocument();
+    expect(screen.getByLabelText('Mail Id')).toBeInTheDocument();
+    expect(screen.getByRole('textbox')).toHaveValue('');
+  });
+
+  it('renderiza con placeholder y value correctos', () => {
+    render(
+      <FormInputWrapper {...defaultProps} placeholder="Enter email" value="test@example.com" />
+    );
+    const input = screen.getByRole('textbox');
+    expect(input).toHaveAttribute('placeholder', 'Enter email');
+    expect(input).toHaveValue('test@example.com');
+  });
+
+  it('dispara onChange con un nuevo valor', async () => {
+    render(<FormInputWrapper {...defaultProps} />);
+    const input = screen.getByRole('textbox');
+    await userEvent.type(input, 'test@example.com');
+    expect(input).toHaveValue('test@example.com');
+    expect(defaultProps.onChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        target: expect.objectContaining({ value: 'test@example.com' }),
+      })
+    );
+  });
+
+  it('no dispara onChange si está disabled', async () => {
+    const handleChange = vi.fn();
+    render(<FormInputWrapper {...defaultProps} onChange={handleChange} disabled />);
+    const input = screen.getByRole('textbox');
+    expect(input).toBeDisabled();
+    await userEvent.type(input, 'test');
+    expect(handleChange).not.toHaveBeenCalled();
+  });
+
+  it('muestra estado de error con errorMessage', () => {
```

Comment:
Test description in Spanish ('muestra estado de error con errorMessage'). Use English. Also the assertion expects aria-invalid attribute; ensure the component explicitly sets aria-invalid on the input when in error state (MUI sets it when error prop is true but if you manipulate inputProps you must ensure it remains).

**Sugerencia:**
```typescript
Translate to English and consider also asserting that helperText is present. Example:

it('shows error state when errorMessage is provided', () => { ... });

And ensure FormInput explicitly sets aria-invalid on the input: inputProps={{ 'aria-label': ..., 'aria-invalid': isError }} or verify MUI behavior in your version.
```